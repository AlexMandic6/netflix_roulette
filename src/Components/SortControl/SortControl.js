import "./SortControl.css";

const SortControl = ({ onSortChange }) => {
	return (
		<div className="sort">
			<label className="sort-by__label" htmlFor="sort-by">
				Sort By:
			</label>
			<div className="select">
				<select
					id="sort-by"
					data-testid="sort-select"
					className="sort-by"
					onChange={onSortChange}
				>
					<option value="release_date">Release Date</option>
					<option value="title">Title</option>
				</select>
			</div>
		</div>
	);
};

export default SortControl;
