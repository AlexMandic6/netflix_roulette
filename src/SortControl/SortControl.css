.select {
	display: grid;
	grid-template-areas: "select";
	align-items: center;
	position: relative;
	min-width: 14ch;
	max-width: 30ch;
	font-size: 1.25rem;
	cursor: pointer;
	line-height: 1.1;
}

select,
&::after {
	grid-area: select;
}

.select::after {
	content: "";
	width: 0.8em;
	height: 0.5em;
	background-color: var(--select-arrow);
	clip-path: polygon(100% 0%, 0 0%, 50% 100%);
	justify-self: end;
}

select,
.select:after {
	grid-area: select;
}

select:focus + .focus {
	position: absolute;
	top: -1px;
	left: -1px;
	right: -1px;
	bottom: -1px;
	border: 2px solid var(--select-focus);
	border-radius: inherit;
}

.sort {
	display: flex;
	align-self: flex-start;
}

.sort-by {
	cursor: pointer;
	margin-left: 30px;
	font-size: 16px;
	font-weight: 500;
	color: white;
	background-color: #232323;
	text-transform: uppercase;
}

.sort-by__label {
	font-size: 16px;
	font-weight: 400;
	color: #a8a8a8;
	text-transform: uppercase;
	white-space: nowrap;
}
