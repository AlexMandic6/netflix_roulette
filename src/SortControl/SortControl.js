import "./SortControl.css";

const SortControl = () => {
	const handleSelectChange = (event) => {
		const selectedValue = event.target.value;
		console.log("Selected option:", selectedValue);
	};

	return (
		<div className="sort">
			<label className="sort-by__label" for="sort-by">
				Sort By:
			</label>
			<div class="select">
				<select
					id="sort-by"
					className="sort-by"
					onChange={handleSelectChange}
				>
					<option value="release-date">Release Date</option>
					<option value="title">Title</option>
				</select>
			</div>
		</div>
	);
};

export default SortControl;
