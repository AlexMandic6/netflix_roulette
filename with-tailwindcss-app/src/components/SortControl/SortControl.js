"use client";
import "./SortControl.css";
// import { useSearchParams } from "react-router-dom";

const SortControl = () => {
	// const [searchParams, setSearchParams] = useSearchParams();
	// const sortBy = searchParams.get("sortBy") || "release_date";

	const handleSortChange = (e) => {
		// setSearchParams((searchParams) => {
		// 	searchParams.set("sortBy", e.target.value);
		// 	return searchParams;
		// });
	};

	return (
		<div className="sort">
			<label className="sort-by__label" htmlFor="sort-by">
				Sort By:
			</label>
			<div className="select">
				<select
					id="sort-by"
					data-testid="sort-select"
					className="sort-by"
					onChange={handleSortChange}
					// value={sortBy}
				>
					<option value="release_date">Release Date</option>
					<option value="title">Title</option>
				</select>
			</div>
		</div>
	);
};

export default SortControl;
