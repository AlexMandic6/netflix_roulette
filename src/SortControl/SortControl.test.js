import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import SortControl from "../SortControl/SortControl";

test("renders SortControl component correctly", () => {
	const { asFragment, getByLabelText } = render(<SortControl />);
	expect(asFragment()).toMatchSnapshot();

	const sortLabel = getByLabelText("Sort By:");
	expect(sortLabel).toBeInTheDocument();
});

test("handles select change correctly", () => {
	const { getByTestId } = render(<SortControl />);
	const selectElement = getByTestId("sort-select");

	fireEvent.change(selectElement, { target: { value: "title" } });
	expect(selectElement.value).toBe("title");

	fireEvent.change(selectElement, { target: { value: "release-date" } });
	expect(selectElement.value).toBe("release-date");
});
