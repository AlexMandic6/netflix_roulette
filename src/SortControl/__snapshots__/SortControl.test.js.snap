// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SortControl component correctly 1`] = `
<DocumentFragment>
  <div
    class="sort"
  >
    <label
      class="sort-by__label"
      for="sort-by"
    >
      Sort By:
    </label>
    <div
      class="select"
    >
      <select
        class="sort-by"
        data-testid="sort-select"
        id="sort-by"
      >
        <option
          value="release-date"
        >
          Release Date
        </option>
        <option
          value="title"
        >
          Title
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
